Packs/CiscoASA/Integrations/CiscoASA/CiscoASA.py

```diff
@@ -1,22 +1,216 @@
+                offset,
+                *args,
+                **kwargs,
+            )
+
+        return wrapper
+
+    def _get_pagination_arguments(
+        self,
+        page: int | None,
+        page_size: int | None,
+        limit: int = DEFAULT_LIMIT
+    ) -> tuple[int, int | None]:
+        """
+        Determine if pagination is automatic or manual and compute remaining items and offset values.
+
+        Args:
+            page (int | None): Page number to return.
+            page_size (int | None): Number of items to return in a page.
+
+        Returns:
+            tuple[bool, int, int | None]:
+                remaining_items: The number of remaining items to fetch.
+                offset: The offset for the next API request.
+        """
+        is_manual = page is not None
+
+        if is_manual:
+            page = page or 1
+            page_size = page_size or self.default_limit
+
+            remaining_items = page_size
+            offset = (page - 1) * page_size + (0 if self.start_count_from_zero else 1)
+        else:
+            remaining_items = limit
+            offset = None
+
+        return remaining_items, offset
+
+    def _handle_pagination(
+        self,
+        client_instance: BaseClient,
+        func: Callable,
+        remaining_items: int,
+        offset: int | None,
+        *args,
+        **kwargs,
+    ) -> tuple[list | dict, list | dict]:
+        """
+        Handle pagination when the API supports both limit and offset parameters.
+
+        Args:
+            client_instance (BaseClient): The instance of the Client class used to call the API request function.
+            func (Callable): API request function to be called.
+            remaining_items (int): The number of remaining items to fetch.
+            offset (int | None): The offset for the next API request.
+            args: Positional arguments to be passed to the API request function.
+            kwarg: Keyword arguments to be passed to the API request function.
+
+        Returns:
+            tuple[list | dict, list | dict]:
+                All the items combined within raw response, All the raw responses combined
+        """
+        kwargs['self'] = client_instance
+
+        raw_items: list[dict[str, Any]] = []
+        raw_responses: list[dict[str, Any]] = []
+
+        # Keep calling the API until the required amount of items have been met.
+        while remaining_items > 0:
+            limit = min(remaining_items, self.api_limit)
+            kwargs |= {
+                'limit': limit,
+                'offset': offset,
+            }
+
+            raw_response = func(
+                *args,
+                **kwargs,
+            )
+
+            raw_item = raw_response
+
+            if self.items_key_path:
+                raw_item = dict_safe_get(raw_item, self.items_key_path)
+
+            if raw_item is None:
+                break
+
+            raw_responses.append(raw_response)
```

Comment:
raw_response should be dict, not array.

Reply:
From the `docstring` of `CommandResults` it can be seen that it accepts a `dict` or `list`.
```
:type raw_response: ``dict`` | ``list``
    :param raw_response: must be dictionary, if not provided then will be equal to outputs. usually must be the original
        raw response from the 3rd party service (originally Contents)
```

The `list` format provides the least modification to the `raw_response` as I only append each `raw_response` in case of multiple calls to the API.
`raw_response` is presented as an item in XSOAR which looks great.
Another thing, this was accepted in 2 PRs in the past (Cisco AMP and Firepower).